File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> mpp; 
        for(int i=0; i<nums.size(); i++){
            int more = target - nums[i];
            if(mpp.find(more) != mpp.end()){
                return {i,mpp[more]};
            }
                mpp[nums[i]] = i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.java
// import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];

        Map<Integer,Integer> helper = new HashMap<>();

        for(int i=0; i<nums.length; i++){
            if(helper.containsKey(target-nums[i])){
                ans[0]=helper.get(target-nums[i]);
                ans[1]=i;
            }else{
                helper.put(nums[i],i);
            }
        }
        return ans;
    }
}